### ISSUE TITLE
`abort()` works, but still get "Warning: Can't perform a React state update on an unmounted component". CodeSandbox provided

### ISSUE THREAD

**Issue by nth-chile** [alex-cory/use-http#207]
When returning `abort` as the `useEffect` cleanup function, the request aborts and the promise resolves as expected, but I still see the React warning: `Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
    in Test (at App.js:15)`

**Question**: How do I cancel a fetch on unmount (and not receive the React warning)?

In case [this Codesandbox link](https://codesandbox.io/s/boring-browser-b7bbm) expires, I'll paste the code below. I tried to keep it simple -- one component unmounts another component before its fetch finishes

App.js - unmount `<Test />` after 100ms (to interrupt the API call made by `<Test />`)
```import React, { useEffect, useState } from "react";
import Test from "./Test";
import "./styles.css";

export default function App() {
  const [bool, setBool] = useState(true);

  // 100ms after mounting, set bool to false
  useEffect(() => {
    setTimeout(() => {
      setBool(false);
    }, 100);
  }, []);

  return <div className="App">{bool && <Test />}</div>;
}
```

Test.js - Make an API call using `useFetch`, and return `abort` from `useEffect`
```import React, { useEffect } from "react";
import useFetch from "use-http";

export default function Test() {
  const { get, abort } = useFetch(
    "https://jsonplaceholder.typicode.com/posts/1"
  );

  const loadIt = async () => await get();

  useEffect(() => {
    loadIt();

    return abort;
  }, []);

  return <h1>I am here</h1>;
}
```

**Comment by alex-cory** [comment]
It should `abort` already when unmounting. [See this line](https://github.com/alex-cory/use-http/blob/master/src/useFetch.ts#L223). I will take a look at this asap

**Comment by nth-chile** [comment]
I see, cool. `abort` does cancel the request, and any statements following the call to `req.get()` will run immediately after `abort` is called. Which kinda makes the React warning seem unwarranted ... or I don't have enough knowledge about unmounting/cleanup

**Comment by alex-cory** [comment]
I think I have a fix for this issue in [this PR](https://github.com/alex-cory/use-http/pull/210/files#diff-326372ad77dca3e7c2460c8d26329f41R54). I block all state from updating if the component is not mounted, so if it still happens I will be very confused.

**Comment by nth-chile** [comment]
Awesome. I will check it out.

**Comment by nth-chile** [comment]
Awesome. I will check it out.

**Comment by alex-cory** [comment]
This should be fixed with `v0.4.0`. Let me know if the issue persists and I'll reopen this.
